server/src/main/kotlin/com/kamelia/jellyfish/plugins/Authentication.kt

```diff
@@ -0,0 +1,32 @@
+package com.kamelia.jellyfish.plugins
+
+import com.auth0.jwt.JWT
+import com.auth0.jwt.algorithms.Algorithm
+import com.kamelia.jellyfish.rest.user.Users
+import com.kamelia.jellyfish.util.Environment
+import io.ktor.server.application.Application
+import io.ktor.server.application.install
+import io.ktor.server.auth.Authentication
+import io.ktor.server.auth.jwt.JWTPrincipal
+import io.ktor.server.auth.jwt.jwt
+
+fun Application.configureAuthentication() {
+    install(Authentication) {
+        jwt("auth-jwt") {
+            realm = Environment.jwtRealm
+
+            val jwtVerifier = JWT.require(Algorithm.HMAC256(Environment.secret)).build()
+            verifier(jwtVerifier)
+
+            validate {credential ->
+                val subject = credential.subject ?: return@validate null
+                val user = Users.findByUsername(subject) ?: return@validate null
+                val lastInvalidation = user.lastInvalidation
+                if (lastInvalidation != null && lastInvalidation.isAfter(credential.issuedAt!!.toInstant())) {
+                    return@validate null
+                }
+                JWTPrincipal(credential.payload)
+            }
+        }
```

Comment:
Don't we need another JWT verifier using the `refreshSecret` to sign and verify the refresh tokens?

```suggestion
        jwt("auth-jwt") {
            realm = Environment.jwtRealm

            val jwtVerifier = JWT.require(Algorithm.HMAC256(Environment.secret)).build()
            verifier(jwtVerifier)

            validate {credential ->
                val subject = credential.subject ?: return@validate null
                val user = Users.findByUsername(subject) ?: return@validate null
                val lastInvalidation = user.lastInvalidation
                if (lastInvalidation != null && lastInvalidation.isAfter(credential.issuedAt!!.toInstant())) {
                    return@validate null
                }
                JWTPrincipal(credential.payload)
            }
        }
        jwt("refresh-jwt") {
            realm = Environment.jwtRealm

            val jwtRefreshVerifier = JWT.require(Algorithm.HMAC256(Environment.refreshSecret)).build()
            verifier(jwtRefreshVerifier)

            validate {credential ->
                // TODO: Validate refresh token
            }
        }
```